RV-Monitor/Kernel_Safety_Claims_by_Runtime_Verification_Monitors.md

```diff
@@ -0,0 +1,542 @@
+
+
+## Monitored Data access
+
+**TODO**: evaluate if, for failure avoidance purposes and hence availability purposes, it is doable to directly read the data to be monitored from the monitored code heap or stack instead of passing it as parameters on the stack. The rationale is that passing critical data through function parameters would increase the surface of possible corruption.
+
+Taking for example an "int" variable, passing it as parameter, either by value or by reference, creates an additional step that is susceptible to interference at runtime.  Using an external reference in the sources + direct read access from the monitoring function, would generate a reference encoded within the executable memory, and thus impervious to accidental overwrites.
+
+
+## Temporal FFI assumption
+
+If it is not possible to guarantee the assumption of “the safety relevant code (including the Kernel code) to be monitored by an external watchdog”, then the FFI temporal interference concept above must be revisited.
+
+
+## How to effectively protect HW or SW resources corruptible from any Kernel code of lower systematic capability
+
+The monitored subsystem is characterized by different operating states where for one or more of such states the correct operation of the system depends on the correct configuration of certain HW or SW resources. \
+Fow such HW or SW resources that are accessible by Kernel code, we should figure out a way to avoid or detect spatial interference.
+
+The interference can manifest itself in different ways:
+
+
+
+* The most obvious is as a direct write operation that affects a memory address region consisting of one or more addresses.
+    * In some cases the addresses may refer to actual physical memory.
+
+        Here, the interference can alter in an unsafe way the content of the affected memory locations.
+
+    * In other cases the addresses may refer to a peripheral device.
+
+        In modern systems the address space used for memory is also used for interacting with certain peripherals (memory mapped I/O), where at certain addresses specified by the processor hardware datasheet it is possible to access both configuration registers and I/O registers of said peripherals.
+
+
+        The effect of the interference is not easy to generalise, because it strongly depends on which type of peripheral and registers are involved and what values might be written there.
+
+    * Especially on 64 bit systems which are capable of addressing a very large address space, it is also possible that nothing is associated with a specific address; in such a case, a write operation will fail with a bus error and it will trigger an exception. This scenario can be considered as safe, because the exception will be noticed by the system and it can be acted upon (usually an MMU translation error).
```

Comment:
Note: exceptions *can* be disabled, in a logic way, even if not through hardware.  For example, the verification that confirms the rodata memory is effectively write protected does just that. It attempts a write over protected memory, but without the kernel treating it as a terminal event. It should be confirmed that there are no chances for exceptions to be disabled, to rely on this assumption.

Reply:
Add a note about care being taken on making sure no drivers disable the safety critical exceptions

Reply:
ok